cmd/dptp-controller-manager/main.go

```diff
@@ -377,39 +398,12 @@ func main() {
 		}
 	}
 
-	allManagers := map[string]controllerruntime.Manager{
-		appCIContextName: mgr,
-		apiCIContextName: registryMgr,
-	}
-	var errs []error
-	for cluster, cfg := range kubeconfigs {
-		if cluster == appCIContextName || cluster == apiCIContextName {
-			continue
-		}
-		if _, alreadyExists := allManagers[cluster]; alreadyExists {
-			logrus.Fatalf("attempted duplicate creation of manager for cluster %s", cluster)
-		}
-		buildClusterMgr, err := controllerruntime.NewManager(cfg, controllerruntime.Options{MetricsBindAddress: "0", LeaderElection: false, DryRunClient: opts.dryRun, Logger: ctrlruntimelog.NullLogger{}})
-		if err != nil {
-			errs = append(errs, fmt.Errorf("failed to construct manager for cluster %s: %w", cluster, err))
-			continue
-		}
-		if err := mgr.Add(buildClusterMgr); err != nil {
-			errs = append(errs, fmt.Errorf("failed to add buildClusterMgr for cluster %s to main mgr: %w", cluster, err))
-			continue
-		}
-		allManagers[cluster] = buildClusterMgr
-	}
-	if err := utilerrors.NewAggregate(errs); err != nil {
-		logrus.WithError(err).Fatal("Failed to construct build cluster managers")
-	}
-
-	allClustersExceptAPICI := map[string]controllerruntime.Manager{}
+	allClustersExceptRegistryCluster := map[string]controllerruntime.Manager{}
 	for cluster, manager := range allManagers {
-		if cluster == apiCIContextName {
+		if cluster == opts.registryClusterName {
 			continue
 		}
-		allClustersExceptAPICI[cluster] = manager
+		allClustersExceptRegistryCluster[cluster] = manager
```

Comment:
You can do this in the first loop where you create the managers

Reply:
Updated.